Add a roadmap page and prioritize mentioning that revenue is not yet implemented

MarkusG opened the pull request:
This resolves #322.

Adding a roadmap to the site not only makes it more available, but the included link also draws more members into the Discord server.

Also, I'm brand new to Vue.js, so any criticism is, of course, welcome.

wafflecoffee commented on layouts/default.vue:
```diff
@@ -35,6 +35,15 @@
+                  'active-path': this.$route.path.startsWith('/mod'),
+                }"
+              >
+                <span>Roadmap</span>
```
This should preferably be in the footer, not the header.

wafflecoffee reviewed (changes requested):
It would also be best for this to target the `apiv2-support` branch rather than the `master` branch - apiv2 has the pending redesign and has a lot of pending changes which may end up conflicting with these changes.

MarkusG commented:
Rebased my changes onto the `apiv2-support` branch, re-styled the roadmap page to match the about page, and also moved the roadmap link from the navbar into the footer.

Geometrically commented:
Sorry for never reviewing this, if you still want this PR, you can retarget it to master. However, the new homepage will have  a roadmap.
